ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -112,16 +112,77 @@ type FetchDecision result = Either FetchDecline result
 -- | All the various reasons we can decide not to fetch blocks from a peer.
 --
 data FetchDecline =
+     -- | The chain is not longer than our chain.  For more details see
+     -- "Ouroboros.Consensus.MiniProtocol.BlockFetch.ClientInterface.mkBlockFetchConsensusInterface"
+     -- which implements 'plausibleCandidateChain'.
+     --
      FetchDeclineChainNotPlausible
+
+     -- | The chain does not intersects with our current chain.
+     --
    | FetchDeclineChainNoIntersection
+
+     -- | The chain has already been fetched.
+     --
    | FetchDeclineAlreadyFetched
+
+     -- | The chain is already being downloaded.
+     --
    | FetchDeclineInFlightThisPeer
```

Comment:
What's the difference between ThisPeer vs OtherPeer?

Reply:
we have two filtering functions:
* `filterNotAlreadyInFlightWithOtherPeers` - in deadline mode it allows to download the same block concurrently from different peers, in block fetch mode it filters out blocks that we are already fetching from other peers
* `filterNotAlreadyInFlightWithPeer` - protects us from starting fetching a block from a peer from which we are already downloading it.

Reply:
I slightly improved the haddocks.

Reply:
Thank you!